lib/rubocop/cop/rspec/scattered_setup.rb

```diff
@@ -56,7 +65,7 @@ def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
         def repeated_hooks(node)
           hooks = RuboCop::RSpec::ExampleGroup.new(node)
             .hooks
-            .select { |hook| hook.knowable_scope? && hook.name != :around }
+            .select { |hook| repeated_hook?(hook) }
```

Comment:
The “repeated hook” is really the whole thing (old lines 56–62), not just what you extracted to a method. I would probably filter twice instead of extracting the method:
```suggestion
            .reject(&:inside_class_method?)
            .select { |hook| hook.knowable_scope? && hook.name != :around }
```

If RuboCop tells you there’s too many lines, or the code gets too complex, feel free to add an inline disable and we’ll refactor later.